### PR TITLE
flatcam: 8.5 -> unstable-2022-02-02

### PR DESCRIPTION
**Pull request by matthiasbeyer**
qt4 -> qt5

(cherry picked from commit 9ab75e0f1ca9cae1d52cf994f7e46c8748e234ba)

---

Extracted from #174634
because this can be PRed independently and can land earlier than the PR above.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/flatcam/default.nix [hunk 1]
```diff
@@ -1,83 +1,98 @@
...
+, pyqt5
+, matplotlib
+, numpy
+, cycler
+, python-dateutil
+, kiwisolver
+, six
+, setuptools
+, dill
+, Rtree
```
Comment (ajs124) [inline pkgs/applications/science/electronics/flatcam/default.nix:13]: apparently this was renamed to `rtree`

**Comment by natsukium** [comment]
```
error: builder for '/nix/store/h1myk825hzv7v99xk6mpjpsarzvly2kp-flatcam-unstable-2022-02-02.drv' failed with exit code 2;
       last 10 log lines:
       > updateAutotoolsGnuConfigScriptsPhase
       > configuring
       > no configure script, doing nothing
       > installing
       > patching script interpreter paths in .
       > ./tests/canvas/prof.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/ir0j7zqlw9dc49grmwplppc7gh0s40yf-bash-5.2-p15/bin/sh"
       > ./assets/linux/flatcam-beta: interpreter directive changed from "#!/bin/bash" to "/nix/store/ir0j7zqlw9dc49grmwplppc7gh0s40yf-bash-5.2-p15/bin/bash"
       > install flags: SHELL=/nix/store/ir0j7zqlw9dc49grmwplppc7gh0s40yf-bash-5.2-p15/bin/bash USER_ID=0 LOCAL_PATH=/build/source/. INSTALL_PATH=/nix/store/8zjb0z5df61r01s99w0y9rkp2mqlsyhq-flatcam-unstable-2022-02-02/share/flatcam APPS_PATH=/nix/store/8zjb0z5df61r01s99w0y9rkp2mqlsyhq-flatcam-unstable-2022-02-02/share/applications install
       > Current python version is 3.10
       > Makefile:22: *** You must have at least 3.5 installed.  Stop.
```
🤔

**Comment by matthiasbeyer** [comment]
Lol, does the makefile handle "3.10" as "3.1"? :laughing:

**Comment by ajs124** [comment]
yup, they do

**Comment by tjni** [comment]
I am pretty sure that I broke `flatcam` on staging-next with the Python bootstrapping changes. Do you think this is close to getting across the finish line, or I can buy more time by merging https://github.com/NixOS/nixpkgs/pull/249849 first.

**Comment by matthiasbeyer** [comment]
Go ahead with your PR, I am not sure what to do about the makefile issue here.

**Comment by natsukium** [comment]
It seems that these lines can be commented out to avoid the version check.
https://bitbucket.org/jpcgt/flatcam/src/ebf5cb9e3094362c4b0774a54cf119559c02211d/Makefile#lines-20:23

**Pull request closed without merging**